docs/stdlib/json.rst

```diff
@@ -91,39 +93,37 @@ expression <select>` in *JSON mode*, including the type shape.
     ...     filter .name = 'std::bool');
     {'{"name": "std::bool", "timestamp": "2019-04-02"}'}
 
-JSON values can also be cast back into scalars. This casting is
-symmetrical meaning that if a scalar can be cast into JSON, only that
-particular JSON type can be cast back into that scalar:
+JSON values can also be casted back as a scalar type. Casting JSON is
+symmetrical: if a scalar type can be casted back into JSON, only that
+particular JSON type can be casted back into that scalar. Some scalar types
+will have specific conditions:
 
-- JSON *string* can be cast into :eql:type:`str`. Casting
-  :eql:type:`uuid` and :ref:`date and time types
+- JSON *strings* can be casted as a :eql:type:`str` type. Casting
+  :eql:type:`uuid` and :ref:`date/time types
   <ref_std_datetime>` to JSON results in a JSON
-  *string* representing the original value. This means that it is
-  also possible to cast a JSON *string* back into these types. The
-  string value has to be properly formatted (much like in case of
-  a :eql:type:`str` value being cast) or else the cast will raise an
-  exception.
-- JSON *number* can be cast into any of
-  the :ref:`numeric types <ref_std_numeric>`
-- JSON *boolean* can be cast into :eql:type:`bool`
-- JSON *null* is special since it can be cast into an ``{}`` of any type
-- JSON *array* can be cast into any valid EdgeDB array, so it must be
-  homogeneous, and must not contain *null*
-
-A *regular* :eql:type:`tuple` is converted into a JSON *array* when cast
-into :eql:type:`json`, whereas a *named* :eql:type:`tuple` is converted
-into a JSON *object*. These casts are not reversible, i.e. it is not
-possible to cast a JSON value directly into a :eql:type:`tuple`.
+  string representing its original value. This means it is also possible to
+  cast a JSON string back to its original type. The value of the string must
+  also be properly formatted, otherwise an exception will be raised.
+- JSON *numbers* can be casted to any :ref:`numeric type <ref_std_numeric>`.
+- JSON *booleans* can be casted as a :eql:type:`bool`.
+- Nullified JSON is special as it can be casted as an empty set or ``{}`` of
+  :eql:type:`anytype`.
+- JSON *arrays* can be casted to any valid EdgeDB array, as long as it's
+  homogeneous and do not contain *null* as a value.
+
+A *regular* :eql:type:`tuple` is converted into a JSON array when casted
+as a :eql:type:`json`, unlike a *named* :eql:type:`tuple` where it is
+converted into a JSON *object*. These casts are not reversible, however,
+i.e. it is not possible to cast a JSON value directly into a
+:eql:type:`tuple`.
```

Comment:
- Leading with the named tuple since it may not be clear what a "regular" or "standard" tuple is if we lead with it. By leading with the named tuple, it's clearer that the other variety of tuple is the one that is not named.
- Slightly prefer "standard" to "regular" although I almost went with "unnamed" here. Open to alternatives.
- Tried to drive home the difference in the tuple to JSON casts with slightly different wording.

```rest
A named :eql:type:`tuple` is converted into a JSON object when cast as a
:eql:type:`json` unlike a standard :eql:type:`tuple` which is converted into a
JSON array. Unlike other casts to JSON, tuple casts to JSON are *not* reversible
(i.e., it is not possible to cast a JSON value directly into a 
:eql:type:`tuple`).
```